src/decode/lzbuffer.rs

```diff
@@ -150,11 +167,21 @@ where
         *self.buf.get(index).unwrap_or(&0)
     }
 
-    fn set(&mut self, index: usize, value: u8) {
-        if self.buf.len() < index + 1 {
-            self.buf.resize(index + 1, 0);
+    fn set(&mut self, index: usize, value: u8) -> error::Result<()> {
+        let new_len = index + 1;
+
+        if new_len > self.memlimit {
+            Err(error::Error::LZMAError(format!(
+                "exceeded memory limit of {}",
+                self.memlimit
+            )))
+        } else {
```

Comment:
Given that this condition is unlikely to be true on good files, can you nest it in the case where resizing is needed?

Reply:
Good catch. I updated the PR.